helper.py

```diff
@@ -490,3 +554,178 @@ def get_mode_velocity(velocity_array_framewise):
+    # input_2 = tf.keras.layers.Input(shape=(64, 1, 10))
+    # input_3 = tf.keras.layers.Input(shape=(64, 1, 10))
+    # input_23 = tf.keras.layers.Concatenate(axis=2)([input_2, input_3])
+    emb = cnn(input)
+    # emb2 = cnn1d(input_23)
+    # emb = tf.keras.layers.Concatenate(axis=1)([emb1, emb2])
+    output = tf.keras.layers.Dense(units=1, activation='linear')(emb)
+    model = tf.keras.Model(inputs=input, outputs=output)
+    # model = tf.keras.Model(inputs=[input_1, input_2, input_3], outputs=output)
+    print(model.summary())
+    # model.compile(loss=SolveEquationLoss(), optimizer='adam', metrics=['mse'])
+    return model
+
+
+def preprocess_input_cnn(X_train):
+    dop_train = np.array(
+        [np.concatenate([np.expand_dims(e, 2) for e in v.transpose(1, 0)[2]], axis=2) for v in X_train])
+    rp_train = np.array(
+        [np.concatenate([np.expand_dims(e.reshape(-1, 1), 2) for e in v.transpose(1, 0)[0]], axis=2) for v in X_train])
+    noiserp_train = np.array(
+        [np.concatenate([np.expand_dims(e.reshape(-1, 1), 2) for e in v.transpose(1, 0)[1]], axis=2) for v in X_train])
+
+    dop_train_s = (dop_train - dop_train.min()) / (dop_train.max() - dop_train.min())
+    rp_train_s = (rp_train - rp_train.min()) / (rp_train.max() - rp_train.min())
+    noiserp_train_s = (noiserp_train - noiserp_train.min()) / (noiserp_train.max() - noiserp_train.min())
+    return dop_train_s, rp_train_s, noiserp_train_s
+
+def traincnn(model, X_train, y_train, epochs=500):
+        X_train = np.asarray(X_train)
+        y_train = np.asarray(y_train)
+        model.compile(loss='mse', optimizer='adam', metrics=["mse"])
+        history = \
+            model.fit(
+                X_train,
+                y_train,
+                epochs=epochs,
+                validation_split=0.2,
+                batch_size=32,
+            )
+        plt.plot(history.history['loss'], label='MSE Loss')
+        plt.plot(history.history['val_loss'], label='Validation MSE Loss')
+        plt.legend()
+        plt.show()
+        return model
+
+
+def train(model, X_train, y_train, L_R_array, epochs=500):
+    X_train = np.asarray(X_train)
+    y_train = np.asarray(y_train)
+    model.compile(loss=SolveEquationLoss(X_train, L_R_array), optimizer='adam', metrics=["mse"])
+    X_train = np.abs(X_train)
+    X_train  = np.sum(X_train, axis=(0,1))
+    X_train = np.expand_dims(X_train, axis = -1)
+    history = \
+        model.fit(
+            X_train,
+            y_train,
+            epochs=epochs,
+            validation_split=0.2,
+            batch_size=32,
+        )
+    plt.plot(history.history['loss'], label='MSE Loss')
+    plt.plot(history.history['val_loss'], label='Validation MSE Loss')
+    plt.plot(history.history['SolveEquationLoss'], label='Combined Loss')
+    plt.plot(history.history['val_SolveEquationLoss'], label='Validation Combined Loss')
+    plt.legend()
+    plt.show()
+    return model
+
```

Comment:
Why 2 train functions. Add only 'mse' loss related train function in this PR.

Reply:
Removed PINN train function